src/components/Charts/mutationsTree.js

```diff
@@ -0,0 +1,48 @@
+
+import React from 'react';
+import Container, {Title} from "./styles";
+
+class MutationsTree extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {};
+    }
+
+    createVariants(variants, level){
+        let variantLines=[];
+        for(const variant of variants){
+            variantLines.push('---'.repeat(level)+"variant: "+ variant.name+" (" + variant.mutations.map(mutation=>mutation.from+mutation.position+mutation.to).join(", ")+")");
+            if(variant.subs.length>0){
+                variantLines=variantLines.concat(this.createVariants(variant.subs, level+1));
+            }
+        }
+        return variantLines;
+    }
+
+
+    componentDidMount() {
+        let variantsTree = this.createVariants(this.props.data, 0).map(i => {
+            return <p>{i}</p>
+        });
+        this.setState({data:variantsTree});      
+    }
+
+    componentDidUpdate(prevProps) {
+        this.render();
```

Comment:
this does not update variants tree if props change. Are we sure we want that? Also, can't this be written in a functional-component with perhaps `useMemo()` hook?

Reply:
I am not sure how to do this :/. I am quite a newbie to react.

Reply:
@ppershing Could we ask you to help us with this, please?

Reply:
I've tried to fix it, somehow